syntax = "proto3";

package booking;

service BookingService {
  rpc BookCar (BookCarRequest) returns (BookCarResponse);
  rpc RestoreAvailability (RestoreAvailabilityRequest) returns (RestoreAvailabilityResponse);
}

message BookCarRequest {
  int32 car_id = 1;
  string start_date = 2;
  string end_date = 3;
  int32 user_id = 4;
}

message BookCarResponse {
  bool success = 1;
  string message = 2;
  float total_price = 3;
}

message RestoreAvailabilityRequest {
  int32 car_id = 1;
  string start_date = 2;
  string end_date = 3;
  float price_per_day = 4;
}

message RestoreAvailabilityResponse {
  bool success = 1;
  string message = 2;
}
